tools/tpm2_encryptdecrypt.c

```diff
@@ -91,29 +108,15 @@ static tool_rc encrypt_decrypt(ESYS_CONTEXT *ectx, TPM2B_IV *iv_start) {
     TPM2B_IV *iv_out = NULL;
     TPM2B_IV *iv_in = iv_start;
 
-    unsigned char pad_data[1] = {0};
-    unsigned i = 0;
-
     while (ctx.input_data_size > 0) {
         ctx.data.size =
             ctx.input_data_size > TPM2_MAX_DIGEST_BUFFER ?
                     TPM2_MAX_DIGEST_BUFFER : ctx.input_data_size;
 
         memcpy(ctx.data.buffer, &ctx.input_data[data_offset], ctx.data.size);
 
-        if (ctx.is_padding_option_enabled && !ctx.is_decrypt &&
-            ctx.data.size < TPM2_MAX_DIGEST_BUFFER &&
-            (ctx.data.size % ctx.padded_block_len)) {
-
-            pad_data[0] =
-                ctx.padded_block_len -(ctx.data.size % ctx.padded_block_len);
-
-            for(i = 0; i < pad_data[0]; i++) {
-                ctx.data.buffer[ctx.data.size + i] = pad_data[0];
-            }
-
-            ctx.data.size += pad_data[0];
-        }
+        uint8_t pad_data = 0;
+        append_pkcs7_padding_data_to_input(&pad_data);
```

Comment:
why not just update the buffer size:
    void pkcs7_add_pad(TPM2B_MAX_DIGEST_BUFFER *buf) {
        
        uint8_t pad = buffer->size % ctx.padded_block_len;
        if (!pad) {
            return;
        }
       
        memset(&buffer->buffer[buffer->size], pad, pad);
        buffer->size += pad;
    }
```C
if ( mode is encrypt) {
    pkcs7_add_pad(&ctx.data)
}
    


```

Reply:
The data buffer probably shouldn't be in global ctx anymore since were reading it block by block from a file source.